File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03331/p03331_num8_parsed.cpp
#include <iostream>
#include <string>

int main() {
    int N;
    std::cin >> N;
    
    int minSum = 999999;
    
    // Iterate through all possible splits of N into two positive integers
    for (int A = 1; A < N; A++) {
        int B = N - A;
        
        // Calculate the sum of digits of A
        int sumA = 0;
        std::string strA = std::to_string(A);
        for (char c : strA) {
            sumA += c - '0';
        }
        
        // Calculate the sum of digits of B
        int sumB = 0;
        std::string strB = std::to_string(B);
        for (char c : strB) {
            sumB += c - '0';
        }
        
        // Update the minimum sum if necessary
        int currSum = sumA + sumB;
        if (currSum < minSum) {
            minSum = currSum;
        }
    }
    
    std::cout << minSum << std::endl;
    
    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03331/p03331_num8_parsed.py
# Read N from input
N = int(input())

# Function to calculate the sum of the digits for a number
def sum_of_digits(num):
    return sum(int(digit) for digit in str(num))

# Initial values for the minimum sum of digits and A
min_sum = float('inf')
A = 1

# Iterate over all possible values of A
while A <= N // 2:
    B = N - A
    sum_A = sum_of_digits(A)
    sum_B = sum_of_digits(B)
    min_sum = min(min_sum, sum_A + sum_B)
    A += 1

# Print the minimum possible value
print(min_sum)
